internal/pkg/scorecard/interfaces.go

```diff
@@ -21,64 +21,71 @@ import (
 	"os"
 	"os/exec"
 	"path/filepath"
+	"strings"
 
+	"github.com/operator-framework/operator-sdk/internal/pkg/scaffold"
 	scplugins "github.com/operator-framework/operator-sdk/internal/pkg/scorecard/plugins"
 	scapiv1alpha1 "github.com/operator-framework/operator-sdk/pkg/apis/scorecard/v1alpha1"
-
-	"github.com/spf13/viper"
+	"github.com/operator-framework/operator-sdk/version"
+	v1 "k8s.io/api/core/v1"
 )
 
 type Plugin interface {
 	Name() string
 	Run() scapiv1alpha1.ScorecardOutput
 }
 
-type genericPlugin struct {
-	binPath string
+type externalPlugin struct {
+	name   string
+	config externalPluginConfig
 }
 
-type internalPlugin struct {
+type basicOrOLMPlugin struct {
 	name       string
 	pluginType scplugins.PluginType
+	config     scplugins.BasicAndOLMPluginConfig
 }
 
-func (p genericPlugin) Name() string {
-	return filepath.Base(p.binPath)
+func (p externalPlugin) Name() string {
+	return p.name
 }
 
-func (p genericPlugin) Run() scapiv1alpha1.ScorecardOutput {
+func (p externalPlugin) Run() scapiv1alpha1.ScorecardOutput {
 	// This should never error since we chdir in getPlugins()
-	if err := os.Chdir(viper.GetString(PluginDirOpt)); err != nil {
-		name := fmt.Sprintf("Failed Plugin: %s", filepath.Base(p.binPath))
-		description := fmt.Sprintf("Plugin with file name `%s` failed", filepath.Base(p.binPath))
+	if err := os.Chdir(filepath.Join(rootDir, scViper.GetString(PluginDirOpt))); err != nil {
+		name := fmt.Sprintf("Failed Plugin: %s", p.name)
+		description := fmt.Sprintf("Plugin with file name `%s` failed", filepath.Base(p.config.Command))
```

Comment:
This is a new problem since we define a name for the plugins with this config design. However, the plugin also returns a name field, and in a successful run, we return the name that the plugin gave us, not the name from the config. This is a bit confusing, because this means we only ever use the config provided name during a failure, where we specify the name. I'm not quite sure how we should handle the name specified in the config file, or if we should even have one at all now...

Reply:
Hmm that's an interesting question. These end up being the suite names right? So going through various options:

1. Plugin provides name in output, config name field is removed
    * Plugin fails to execute or does not return name in output => use basename of plugin executable
    * Plugin returns name in output => use name provided by plugin
2. Plugin provides name, but can be overridden by config
    * Plugin fails to execute or does not return name in output => use config name if present, fallback to basename of plugin executable
    * Plugin returns name in output => use config name if present, fallback to name provided by plugin
3. Plugin name is always basename of plugin executable
4. Plugin name is always provided by config

None of these are perfect IMO.

Option 1 gives control to the plugin author. Plugin authors *could* do something weird, but generally I'd imagine this would be static or based on flags/input. This property is nice because shared plugins will generally result in the same name when run against different projects, which makes it easier to compare apples-to-apples.

Option 2 is probably my least favorite option, since its the \\\_(ツ)\_/ of the group.

Option 3 and 4 are nice because duplicates can be found without executing the plugins, and they're very straightforward. Not sure if/how we handle duplicate names? Con: the same plugin might be renamed or configured with a different name, making comparison across projects a no-go.

I think I lean towards option 1. Thoughts? Any other options I'm not thinking of?

Reply:
A plugin could return multiple suites. This would mean that there would be multiple names (each suite has a name), so changing the name(s) of the suite using the scorecard config can get messy.

Option 1 is good, but there is the issue that people may want to run the same plugin with different arguments or environment. If we only print the command's basename, a user may have difficulty identifying which test failed. By also providing the name that a user provided in their config, it makes it a lot easier to identify which plugin/configuration failed.

Reply:
We could go with option 1 and append an `md5` digest of all arguments/environment data to the plugin name, so you can run the same plugin with different configurations.

Reply:
> A plugin could return multiple suites. This would mean that there would be multiple names (each suite has a name), so changing the name(s) of the suite using the scorecard config can get messy.

Maybe having a plugin return multiple suites is worth revisiting in scorecard 2.0. In the current output, is it possible to know which suites came from which plugin? If plugins ran only one suite, the plugin name would be the suite name, which seems like a decent way to handle plugins.

Either way, that isn't the case now. So in the meantime, it sounds like we're in agreement on not allowing user-overrideable test names?

> Option 1 is good, but there is the issue that people may want to run the same plugin with different arguments or environment. If we only print the command's basename, a user may have difficulty identifying which test failed. By also providing the name that a user provided in their config, it makes it a lot easier to identify which plugin/configuration failed.

True, but it sounds like the only time the basename would be used is when the plugin fails. In that case, can we populate the log output with the full path, args, env of the failed plugin, along with whatever the stdout/err was.